FirebaseAuth/Sources/Swift/Utilities/AuthDefaultUIDelegate.swift

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if os(iOS) || os(tvOS)
+
+  import Foundation
+  import UIKit
+  @_implementationOnly import GoogleUtilities
+
+  /** @class AuthDefaultUIDelegate
+      @brief Class responsible for providing a default FIRAuthUIDelegte.
+      @remarks This class should be used in the case that a UIDelegate was expected and necessary to
+          continue a given flow, but none was provided.
+   */
+  class AuthDefaultUIDelegate: NSObject, AuthUIDelegate {
+    // TODO: Figure out what to do for extensions.
+    /** @fn defaultUIDelegate
+        @brief Returns a default FIRAuthUIDelegate object.
+        @return The default FIRAuthUIDelegate object.
+     */
+    @available(iOSApplicationExtension, unavailable)
+    @available(tvOSApplicationExtension, unavailable)
+    class func defaultUIDelegate() -> AuthUIDelegate? {
+      // iOS App extensions should not call [UIApplication sharedApplication], even if UIApplication
+      // responds to it.
+      guard let applicationClass = NSClassFromString("UIApplication"),
+            applicationClass.responds(to: NSSelectorFromString("sharedApplication")) else {
+        return nil
+      }
```

Comment:
I see you have the TODO to refine it, but I'm curious does this currently gives you? Do lines 32 and 33 make the guard statement obsolete?

Reply:
Probably, but I left it in the context of the TODO and unclear documentation when searching.  More thorough extension testing is needed.

Reply:
Added a more clear todo